core/common/src/main/java/alluxio/underfs/ObjectUnderFileSystem.java

```diff
@@ -1118,14 +1118,25 @@ protected boolean parentExists(String path) throws IOException {
    */
   @VisibleForTesting
   public String stripPrefixIfPresent(String path) {
-    final String normalizedPath = PathUtils.normalizePath(path, PATH_SEPARATOR);
     String stripedKey = CommonUtils.stripPrefixIfPresent(
-        normalizedPath,
+        path,
         PathUtils.normalizePath(mRootKeySupplier.get(), PATH_SEPARATOR));
-    if (!stripedKey.equals(normalizedPath)) {
+    if (!stripedKey.equals(path)) {
       return stripedKey;
     }
-    return CommonUtils.stripPrefixIfPresent(normalizedPath, PATH_SEPARATOR);
+
+    // See if adding a PATH_SEPARATOR makes a difference in stripping prefix
```

Comment:
What is an example where this will make a difference? `PathUtils.normalizePath` just adds a separator at the end, so does this only apply to the single string `s3://bucket`?

Reply:
That is correct.